src/main/java/org/broadinstitute/consent/http/service/DarCollectionService.java

```diff
@@ -75,29 +84,227 @@ public DarCollectionService(DarCollectionDAO darCollectionDAO, DarCollectionServ
+          .format(createDate).toString();
+        summary.setDarCode(darCode);
+        summary.setStatus(DarCollectionStatus.DRAFT.getValue());
+        summary.setName(d.getData().getProjectTitle());
+        summary.addAction(DarCollectionActions.RESUME.getValue());
+        summary.addAction(DarCollectionActions.DELETE.getValue());
+        summaries.add(summary);
+      } catch(Exception e) {
+        logger.warn("Error processing draft with id: " + d.getId());
+      }
+    });
+  }
+
+  private void processDarCollectionSummariesForResearcher(List<DarCollectionSummary> summaries) {
+    //if an election exists, cancel does not appear
+    //if there are no elections, review and cancel are present
+    //if the collection is canceled, revise and review is present
+    Map<String, Integer> statusCount = new HashMap<>();
+    summaries.forEach(s -> {
+      Map<Integer, Election> elections = s.getElections();
+      int electionCount = elections.values().size();
+      elections.values().forEach(election -> updateStatusCount(statusCount, election.getStatus()));
+      s.addAction(DarCollectionActions.REVIEW.getValue());
+      //check dar statuses, if they're all canceled show revise (but only if there are no elections)
+      Boolean isCanceled = false;
+      if(electionCount == 0) {
+        Collection<String> darStatuses = s.getDarStatuses().values();
+        isCanceled = darStatuses.size() > 0 && darStatuses.stream()
+          .allMatch(st -> st.equalsIgnoreCase(DarStatus.CANCELED.getValue()));
+        if(isCanceled) {
+          s.addAction(DarCollectionActions.REVISE.getValue());
+          s.setStatus(DarCollectionStatus.CANCELED.getValue());
+        } else {
+          s.addAction(DarCollectionActions.CANCEL.getValue());
+          s.setStatus(DarCollectionStatus.UNREVIEWED.getValue());
+        }
+      } else {
+        determineCollectionStatus(s, statusCount, s.getDatasetCount(), s.getElections().size());
+      }
+    });
+  }
+
+  private void processDarCollectionSumariesForMember(List<DarCollectionSummary> summaries) {
+    summaries.forEach(s -> {
+      Collection<Election> elections = s.getElections().values();
+        Integer electionCount = elections.size();
+        //if there are no elections present, unreviewed
+        //if there are elections present. in process
+        if(electionCount == 0) {
+          s.setStatus(DarCollectionStatus.UNREVIEWED.getValue());
+        } else {
+          Boolean isVotable = elections
+            .stream()
+            .anyMatch(election -> election.getStatus().equalsIgnoreCase(ElectionStatus.OPEN.getValue()));
+          
+          if(isVotable) {
+            s.setStatus(DarCollectionStatus.IN_PROCESS.getValue());
+            s.addAction(DarCollectionActions.VOTE.getValue());
+          } else {
+            //non-votable states
+              //all canceled (complete)
+              //some datasets do not have elections (in process) (legacy)
```

Comment:
nit: This case can happen for both legacy and new DARs so I don't think we need to limit this (comment-wise) to legacy DARs. For new DARs, this can happen when it is across two DACs, one dac having created elections and the other DAC not having created elections.